### PR TITLE
chore: trim unused STL includes

### PR DESCRIPTION
**Pull request by dsanders11**
#### Description of Change

This PR removes unused `#include`s for STL headers.

There's a handful of added lines in `*.cc` files which transitively relied on an `#include` which was trimmed.

#### Checklist

- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes

#### Release Notes

Notes: none

### CODE REVIEW

**Review by deepak1556 — APPROVED** [review]
was iwyu used to trim these ?

**Comment by dsanders11** [comment]
> was iwyu used to trim these ?

Nope, just good old manual work. Would be great if it were feasible to get the IWYU tool to work on the codebase. Chromium themselves seem to have gotten mixed results trying to get it to work, lots of edge cases which confuse it seems. So after reading that I didn't pursue that approach as it seemed like it might be a waste of time to get it built and working with the code base. It would certainly be helpful though, I have another WIP change which trims other unused `#include`s, but it's a lot more time consuming since it's not as straight forward. I think there's another around another 100 (maybe more?) unused includes which could be trimmed that are non-STL.

Basically for this PR I used VS Code's handy search features. Searched files for `#include <` (500+ results or so), then left a standing search for something like `std::map` and clicked through all the results for `#include <map>` and trimmed any which didn't have any search results. It's nice that you can clear individual search results in VS Code. Similar approach for other headers until I was out of results.

**Comment by dsanders11** [comment]
PR updated, found a few more I missed.

**Review by zcbenz — APPROVED** [review]
